src/redux/sagas/expenditures/createStakedExpenditure.ts

```diff
@@ -119,6 +119,7 @@ function* createStakedExpenditure({
       });
     }
 
+    yield takeFrom(approveStake.channel, ActionTypes.TRANSACTION_CREATED);
```

Comment:
@jakubcolony FYI, waiting for the tx to be created before you start altering its status is necessary to avoid the situation as I just discovered where pending/ready fire first and then the saga chain breaks.

Reply:
It's probably worth checking all expenditure sagas follow this pattern.

Reply:
Good shout, tracked here: https://github.com/JoinColony/colonyCDapp/issues/1098